### ISSUE TITLE
Decimal precision issue

### ISSUE THREAD

**Issue by chamoda** [bigcapitalhq/bigcapital#203]
Thanks for this app.

I've noticed that decimal fractional values get rounded into integers and this makes some reports inaccurate. 

Here's the steps to recreate.

1. Create an item with price 100.50
2. Create an invoice with the item
3. Go to Journal sheet. Debit is 100.50 but Credit is 101.00 with .50 cents difference

![Screenshot from 2023-08-02 15-11-01](https://github.com/bigcapitalhq/bigcapital/assets/2217493/1308de83-aee2-47c5-b2bc-177d77be5063)

**Comment by asenawritescode** [comment]
@abouolia 

This is odd. Did you fix this issue ?
I am unable to replicate it.

![image](https://github.com/bigcapitalhq/bigcapital/assets/67445192/9cc34ab8-3134-4444-a1bd-16c5192528ca)

**Comment by xprnio** [comment]
I'm having the same issue.
I created an invoice which has an item with a rate of 13 EUR and a quantity of 159,58 which should total 2.074,54 but instead it rounds it up to 2.080,00

![image](https://github.com/bigcapitalhq/bigcapital/assets/3042904/4dc9d4ca-4298-4c97-8e0a-498a8c4d9632)

**Comment by xprnio** [comment]
My issue was caused by the `quantity` field in `entries` being an `int` instead of a `decimal`.
I'll send a PR up tomorrow for changing it to a `decimal`

**Comment by abouolia** [comment]
@xprnio You're absolutely right, when I wrote the database migration never thought it could be `1.123457` I thought would be integer all the time, I'd be grateful if you created a PR to fix it, btw here's some things in my mind to do to fix it.

- Create a new database migration on tenants DBs and there're some especial commands for Bigcapital, you can create a new one by `node packages/server/build/commands.js tenants:migrate:make convert_items_entries_quantity_to_decimal` if the `commands.js` file wasn't exist run `pnpm run build:server` to build it.
- Change the controllers  validation schema of of the quantity of all services that have `quantity`, I remember the services have quantity invoices, receipts, estimates, credit notes, bills and vendor credits.

If you have any more questions please let me know.

**Comment by xprnio** [comment]
@abouolia I just created the PR, however I didn't run the e2e tests locally (running Arch with Brave and I couldn't figure out how to configure Playwright to use that), but I saw the GH Actions also run those. So now I'm just waiting for authorization for it to run I guess.

You can find the PR here:
https://github.com/bigcapitalhq/bigcapital/pull/316
